### PR TITLE
NEW Archive admin to manage archived records

### PR DESCRIPTION
**Pull request by lukereative**
Allows row to be clicked to view record if a view action present.

**Implements:** https://github.com/silverstripe/silverstripe-framework/issues/6202

**Sibling PRs:**
- https://github.com/silverstripe/silverstripe-framework/pull/8240
- https://github.com/silverstripe/silverstripe-versioned/pull/168
- https://github.com/silverstripe/silverstripe-versioned-admin/pull/29

### CODE REVIEW

**Comment by maxime-rainville** [comment]
The OK/Cancels buttons on the URL segment don't do anything.

image

The BreadCrumb when viewing a specific item says `Pages`
image

**Comment by maxime-rainville** [comment]
How much thought as their been put on how to handle Versioned objects that don't use stage publishing?

File: code/ArchiveAdmin.php [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+{
+    private static $url_segment = 'archive';
+
+    private static $menu_title = 'Archives';
+
+    private static $menu_icon_class = 'font-icon-box';
+
+    public $showSearchForm = false;
+
+    public $showOthers = false;
```
[the remark this answers is not in the document]
Comment (lukereative) [reply code/ArchiveAdmin.php]: $showSearchForm disables the default model admin search 

$showOthers has been removed

[the remark this answers is not in the document]
Comment (maxime-rainville) [reply code/ArchiveAdmin.php]: Interesting ... these looks like they should be config flags and be define as `private static`. Do you know why ModelAdmin as a bunch of public attributes?

File: client/src/legacy/ArchiveAdmin.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+$.entwine('ss', ($) => {
+  $('.ArchiveAdmin__action--restore').entwine({
+    onmatch() {
+      $(this).attr('readonly', false);
+      $(this).attr('disabled', false);
+    }
+  });
+
+  $('.other-model-selector select').entwine({
```
Comment (chillu) [inline client/src/legacy/ArchiveAdmin.js]: These should always be scoped to the section they're applying to (`.ArchiveAdmin` or something). It's legacy, but let's not make legacy worse ;)

**Comment by lukereative** [comment]
Closing as code has moved to versioned-admin, new PR here: https://github.com/silverstripe/silverstripe-versioned-admin/pull/29

**Comment by lukereative** [comment]
There's actually, still small, relevant changes regarding the GridField view action

**Pull request merged**